docs/ruleset_functions/get_energy_required_to_heat_swh_use.md

```diff
@@ -0,0 +1,89 @@
+## get_energy_required_to_heat_swh_use
+
+Description: This function calculates the total energy required to heat the SWH use over the course of a year in btu.  Note - this function does not work for service water heating uses with use_units == "OTHER".  In this case, it will return 0.
+
+Inputs:
+- **swh_use**
+- **RMD**
+
+Returns:
+- **energy_required_by_space**: A dict where they keys are space_ids and values are the total energy required to heat the swh_use for that space.  If a swh_use is not assigned to any spaces, the key will be "NO_SPACES_ASSIGNED"
+
+Function Call:
+
+- get_spaces_served_by_SWH_use
+- get_obj_by_id
+
+Data Lookup: None
+
+Logic:
+
+- get the schedule: `hourly_schedule = swh_use.use_multiplier_schedule`
+- get the distribution system: `distribution_system = swh_use.served_by_distribution_system`
+- get the temperature at fixture: `supply_temperature = distribution_system.design_supply_temperature`
+- get the water used: `swh_use_value = swh_use.use`
+- get the inlet water temperature: `inlet_T_sched = distribution_system.entering_water_mains_temperature_schedule`
+- set the drain heat recovery efficiency to 0: `drain_heat_recovery_efficiency = 0`
+- get the drain heat recovery efficiency if heat is recovered by drain: `if(swh_use.is_heat_recovered_by_drain): drain_heat_recovery_efficiency = distribution_system.drain_heat_recovery_efficiency`
+- get the use_units: `use_units = swh_use.use_units`
+
+- now we need to get all of the spaces that the swh_use is applied to.  The rule is that if spaces reference the swh_use, then the use applies to only those spaces.  If the use references no spaces, it applies to all spaces in the building segment.
```

Comment:
The convention is that if any spaces reference the swh_use, then the service water heating use applies to only those spaces. If no spaces reference the service water heating use, it applies to all spaces in the building segment.